core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -353,10 +353,15 @@ class DAGScheduler(
     if (mapOutputTracker.containsShuffle(shuffleDep.shuffleId)) {
       val serLocs = mapOutputTracker.getSerializedMapOutputStatuses(shuffleDep.shuffleId)
       val locs = MapOutputTracker.deserializeMapStatuses(serLocs)
-      for (i <- 0 until locs.length) {
-        stage.outputLocs(i) = Option(locs(i)).toList // locs(i) will be null if missing
+      var numAvailableOutputs = 0
+      var i = 0
+      while (i < locs.length) {
+        if (locs(i) ne null) {
```

Comment:
Ok I used the "ask-Shivaram" algorithm to determine whether this was standard.  But fine to leave it.